### PR TITLE
Convert existing BPR to ruleset with merge queue

### PR DESCRIPTION
**Pull request by netomi**
As discussed in https://github.com/eclipse-csi/otterdog/issues/86.

### CODE REVIEW

**Comment by netomi** [comment]
Actually, now that merge queues are also supported by the ECA validation service, we can also re-add that status check.

**Comment by netomi** [comment]
the default config has been updated to be aligned with the default settings for a branch protection rule. Now the conversion should be much easier.

Furthermore, I grouped now settings related to requiring a pull request into an embedded object to make sense easier to understand and configure, similar to what has been done for merge queues.

File: otterdog/eclipse-apoapsis.jsonnet [hunk 1]
```diff
@@ -72,7 +72,7 @@ orgs.newOrg('eclipse-apoapsis') {
           ],
           requires_pull_request: true,
           required_approving_review_count: 1,
-          required_status_checks: [
+          required_status_checks+: [
```
Comment (mnonnenmacher) [inline otterdog/eclipse-apoapsis.jsonnet:76]: https://gitlab.eclipse.org/eclipsefdn/it/api/git-eca-rest-api/-/issues/161 is not yet closed, is the ECA check already executed by the merge queue?

Comment (netomi) [reply otterdog/eclipse-apoapsis.jsonnet:76]: I pinged the project about the status as I am also not aware tbh.

Comment (mnonnenmacher) [reply otterdog/eclipse-apoapsis.jsonnet:76]: I will also check when I see the next PR in our merge queue.

Comment (mnonnenmacher) [reply otterdog/eclipse-apoapsis.jsonnet]: Thanks, I also just confirmed that it is not yet executed in the merge queue.

**Review by mnonnenmacher — APPROVED** [review]

**Comment by netomi** [comment]
Merge Queue has been correctly applied:

image

**Comment by mnonnenmacher** [comment]
> the structuring of the settings also is much nicer with rulesets, backporting this to branch protection rules will be some work as we would need to adjust all existing configurations.

To me there is no need to backport this if rulesets are the future anyway.

**Pull request merged**